frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segment_statistics_modal.tsx

```diff
@@ -0,0 +1,171 @@
+import { getSegmentBoundingBoxes, getSegmentVolumes } from "admin/admin_rest_api";
+import { Modal, Spin, Table } from "antd";
+import saveAs from "file-saver";
+import { formatNumberToUnit, formatNumberToVolume } from "libs/format_utils";
+import { useFetch } from "libs/react_helpers";
+import { Unicode, Vector3 } from "oxalis/constants";
+import { getResolutionInfo } from "oxalis/model/accessors/dataset_accessor";
+import { Segment } from "oxalis/store";
+import React from "react";
+import { TreeNode } from "./segments_view_helper";
+import { api } from "oxalis/singletons";
+
+const SEGMENT_STATISTICS_CSV_HEADER =
+  "segmendId,segmentName,groupId,groupName,volumeInVoxel,volumeInNm3,boundingBoxTopLeftPositionX,boundingBoxTopLeftPositionY,boundingBoxTopLeftPositionZ,boundingBoxSizeX,boundingBoxSizeY,boundingBoxSizeZ";
+
+const { ThinSpace } = Unicode;
+
+type Props = {
+  onCancel: (...args: Array<any>) => any;
+  isOpen: boolean;
+  tracingId: any;
+  tracingStoreUrl: any;
+  visibleSegmentationLayer: any;
+  relevantSegments: Segment[];
+  parentGroup: number;
+  groupTree: TreeNode[];
+};
+
+type SegmentInfo = {
+  segmentId: number;
+  segmentName: string;
+  groupId: number | undefined | null;
+  groupName: string;
+  volumeInNm3: number;
+  formattedSize: string;
+  volumeInVoxel: number;
+  boundingBoxTopLeft: Vector3;
+  boundingBoxPosition: Vector3;
+};
+
+const exportStatisticsToCSV = (
+  segmentInformation: Array<SegmentInfo>,
+  tracingId: string,
+  groupId: number,
+) => {
+  if (segmentInformation.length < 0) {
+    return;
+  }
+  const segmentStatisticsAsString = segmentInformation
+    .map(
+      (segmentInfo) =>
+        `${segmentInfo.segmentId},${segmentInfo.segmentName},${segmentInfo.groupId},${segmentInfo.groupName},${segmentInfo.volumeInVoxel},${segmentInfo.volumeInNm3},${segmentInfo.boundingBoxTopLeft},${segmentInfo.boundingBoxPosition}`,
+    )
+    .join("\n");
+  const csv = [SEGMENT_STATISTICS_CSV_HEADER, segmentStatisticsAsString].join("\n");
+  const filename = `segmentStatistics_tracing-${tracingId}_group-${groupId}.csv`;
+  const blob = new Blob([csv], {
+    type: "text/plain;charset=utf-8",
+  });
+  saveAs(blob, filename);
+};
+
+export function SegmentStatisticsModal({
+  isOpen,
+  onCancel,
+  tracingId,
+  tracingStoreUrl,
+  visibleSegmentationLayer,
+  relevantSegments: segments,
+  parentGroup,
+  groupTree,
+}: Props) {
+  const mag = getResolutionInfo(visibleSegmentationLayer.resolutions);
+  const nmFactorToUnit = new Map([[1, "nm³"]]);
+  const dataSource = useFetch(
+    async () => {
+      await new Promise((resolve) => setTimeout(resolve, 10000)); //TODO delete; only for testing
+      await api.tracing.save();
```

Comment:
TODO: maybe omit

Reply:
I think this is the best solution if there is no way clear way to determine whether there are unsaved changes

Reply:
(which I think there isnt because we could have used it in the segment context menu in the viewport)

Reply:
actually this doesnt seem to work. maybe we can look into this together once we have the time?